tests/unit/progress/test_workflows.py

```diff
@@ -47,22 +48,31 @@ def test_migration_progress_runtime_refresh(run_workflow, task, crawler, history
 def test_migration_progress_runtime_tables_refresh(run_workflow) -> None:
     """Ensure that the split crawl and update-history-log tasks perform their part of the refresh process."""
     mock_tables_crawler = create_autospec(TablesCrawler)
+    mock_migration_status_refresher = create_autospec(TableMigrationStatusRefresher)
     mock_history_log = create_autospec(ProgressEncoder)
     context_replacements = {
         "tables_crawler": mock_tables_crawler,
+        "migration_status_refresher": mock_migration_status_refresher,
         "tables_progress": mock_history_log,
         "named_parameters": {"parent_run_id": 53},
     }
 
-    # The first part of a 2-step update: the crawl without updating the history log.
+    # The first part of a 3-step update: the table crawl without updating the history log.
     run_workflow(MigrationProgress.crawl_tables, **context_replacements)
     mock_tables_crawler.snapshot.assert_called_once_with(force_refresh=True)
+    mock_tables_crawler.snapshot.reset_mock()
```

Comment:
Did not know this, nice!

Do you prefer a reset over separate tests?

Reply:
No, I would normally say we should _never_ reset mocks.

However in this test we are checking a sequence of operations that need to work together, and resetting allows the test to verify the interactions are happening at the correct point (rather than accumulating).

This _could_ be split into a sequence of tests (one for each task), but grouping them here in this way emphasises that the tasks are intended to run in a particular sequence.

Reply:
I just realised that the test did nothing to verify the task ordering, so I've added that in.

With that in mind, what would you prefer: like this, or split into separate unit tests? I'm fine with either approach…

Reply:
Generally I prefer separate unit tests over big unit tests